### PR TITLE
completed es6-classes exercise.

### PR DESCRIPTION
**Pull request by RossBrewster**
Screenshot 2023-09-26 at 4 15 21 PM
Screenshot 2023-09-26 at 4 15 07 PM
Screenshot 2023-09-26 at 4 15 14 PM

### CODE REVIEW

File: es6-classes/circle.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Circle */
+class Circle extends Shape {
+  constructor(radius) {
+    super(radius);
+    this.radius = radius;
+    this.area = Math.PI * radius ** 2;
+    this.perimeter = 2 * Math.PI * radius;
```
Comment (shawnkost) [inline es6-classes/circle.js]: You shouldn't be creating these properties manually. You should be utilizing `super` to generate these properties for you

File: es6-classes/square.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-undef -- Shape is imported by .html */
+/* exported Square */
+class Square extends Shape {
+  constructor(width) {
+    super(width);
+    this.width = width;
+    this.area = width * width;
+    this.perimeter = width * 4;
```
Comment (shawnkost) [inline es6-classes/square.js]: You shouldn't be creating these properties manually. You should be utilizing `super` to generate these properties for you

**Review by shawnkost — CHANGES_REQUESTED** [review]
Couple changes here Ross!

**Comment by RossBrewster** [comment]
Is this what you mean?

File: es6-classes/circle.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ class Circle extends Shape {
   constructor(radius) {
     super(radius);
     this.radius = radius;
-    this.area = Math.PI * radius ** 2;
-    this.perimeter = 2 * Math.PI * radius;
+    super.area = Math.PI * radius ** 2;
+    super.perimeter = 2 * Math.PI * radius;
```
Comment (shawnkost) [inline es6-classes/circle.js]: Not quite! You're not setting the `area`  & `perimeter` properties directly on `super`. You're supposed to call `super` and pass in the data that you want `area` & `perimeter` to be. When you do `super()` you are calling your `Shape` constructor function, which will generate the `area` & `perimeter` properties for you. So you just need to pass data into that function, to set it's values

**Review by shawnkost — CHANGES_REQUESTED** [review]
Not quite Ross! Left a comment below but it applies to both files

**Comment by RossBrewster** [comment]
Third time's the charm.

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**